Packs/GsuiteAuditor/Integrations/GsuiteAuditor/GsuiteAuditor.py

```diff
@@ -0,0 +1,344 @@
+        if ADMIN_EMAIL:
+            suffix = URL_SUFFIX.format('all', 'admin')
+            client.http_request(url_suffix=suffix, method='GET', params={'max_results': '1'})
+        else:
+            return_results("Please insert Admin Email parameter for the test to run")
+    return 'ok'
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def activities_list_command(client: GSuiteClient, args: Dict[str, Any]) -> CommandResults:
+    """
+    Prints all activities in the G Suite instance.
+
+    :param client: client object which is used to get response from api
+    :param args: command arguments.
+
+    :return: CommandResults object with context and human-readable.
+    """
+    arguments = prepare_args_for_activities_list(args)
+    response = client.http_request(
+        url_suffix=URL_SUFFIX.format(args.get('user_key', 'all'), args.get('application_name')),
+        params=arguments)
+
+    # Readable Output
+    readable_items = prepare_readable_items(response.get('items', []))
+
+    readable_output = f'### Next Page Token: {response["nextPageToken"]}\n' if response.get("nextPageToken") else ''
+    readable_output += tableToMarkdown(
+        'Total Retrieved {}: {}'.format('Activities', len(response.get('items', []))),
+        readable_items,
+        ['time', 'applicationName', 'email', 'profileId', 'ipAddress', 'events']) if response.get(
+        'items') else 'No activities found for the given ' \
+                      'argument(s). '
+    # Context
+    context_outputs = prepare_output_for_activities_list(response)
+
+    return CommandResults(
+        outputs=context_outputs,
+        readable_output=readable_output,
+        raw_response=response
+    )
+
+
+@logger
+def fetch_incidents(client: GSuiteClient, first_fetch_time: str, fetch_limit: int, applications: Any) -> List[Dict]:
+    total_items = []
+    last_run_dict = demisto.getLastRun()
+    last_run = last_run_dict.get('last_run')
+    last_ids = last_run_dict.get('last_ids', [])
+    last_leftover_items = last_run_dict.get('last_leftover_items', [])
+    new_leftover_items = []
+    if not last_run:  # this is the first run
+        last_run = dateparser.parse(first_fetch_time).strftime(DATE_FORMAT)
+
+    end_time = create_end_time(last_run, "1 hours")
+
+    if len(last_leftover_items) != 0:
+        total_items = last_leftover_items[:fetch_limit]
+        new_leftover_items = last_leftover_items[fetch_limit:]
+
+    if len(total_items) < fetch_limit:
+        for app in applications:
+            total_items.extend(fetch_request(client, app, last_run, end_time))
+
+    sorted_items = sorted(total_items, key=lambda k: k['id']['time'])  # sort the data from earlist to last.
+
+    incidents = []
+    counter = 0
+    for i in range(0, len(sorted_items)):
+        item = sorted_items[i]
+        if counter == fetch_limit:
+            new_leftover_items.extend(sorted_items[i:])
+            break
+
+        if item['id']['uniqueQualifier'] in last_ids:
+            continue
```

Comment:
I guess this is to prevent getting the same incident twice? add a dev comment.

Reply:
it is, added a comment